cylc/flow/network/schema.py

```diff
@@ -867,330 +905,677 @@ class Meta:
     res = await resolvers.nodes_mutator(info, command, ids, w_args, args)
     return GenericResponse(result=res)
 
+# Input types:
 
-# Mutations defined:
-class ClearBroadcast(Mutation):
-    class Meta:
-        description = """Expire all settings targeting cycle points
-earlier than cutoff."""
-        resolver = mutator
 
-    class Arguments:
-        workflows = List(String, required=True)
-        command = String(default_value='clear_broadcast')
-        point_strings = List(
-            String,
-            description="""`["*"]`""",
-            default_value=['*'])
-        namespaces = List(
-            String,
-            description="""namespaces: `["foo", "BAZ"]`""",)
-        cancel_settings = List(
-            GenericScalar,
-            description="""
-settings: `[{environment: {ENVKEY: "env_val"}}, ...]`""",)
+class WorkflowID(String):
+    """A registered workflow."""
 
-    result = GenericScalar()
 
+class CyclePoint(String):
+    """An integer or date-time cyclepoint."""
+
+
+class CyclePointGlob(String):
```

Comment:
I suppose all this subclassing of scalar types is so you can centralise the introspect-ability to the type, and hence have the liberty to call the argument whatever you like (as a opposed to always calling all CPGlobs `cycle_point = String(description="""...""")`) and not repeat descriptions.

Reply:
Yeah, it's basically class names as interfaces.

To a GraphQL introspection query a String looks like this:

```
{
    name: 'String',
    kind: 'SCALAR',
    ofType: null
}
```

Whereas a CyclePointGlob looks like this:

```
{
    name: 'CyclePointGlob',
    kind: 'SCALAR',
    ofType: null
}
```

This allows us to differentiate in the UI and use a mapping there to find an appropriate component to render the field.